modules/core/src/main/java/org/apache/ignite/internal/processors/cache/query/GridCacheQueryManager.java

```diff
@@ -869,36 +852,6 @@ private GridCloseableIterator scanIterator(final CacheQuery<?> qry, IgniteClosur
             return iter;
         }
         catch (IgniteCheckedException | RuntimeException e) {
-            if (intFilter != null)
```

Comment:
qry.scanFilter() is not closed now (for example, when we cant reserve locPart)

Reply:
scanFilter are not closeable.
The only reason we have `close` method is to invoke 
```
            if (scanFilter instanceof PlatformCacheEntryFilter)
                ((PlatformCacheEntryFilter)scanFilter).onClose();
```

For now, we allocate any resources only, when locPart reserved, already.

Reply:
Good catch. Fixed. Thanks.